padinfo/boardgenerator.py

```diff
@@ -0,0 +1,88 @@
+class BoardGenerator(object):
+    """Generate a Dawnglare link with optional inversion."""
+
+    allowed_letters = "RGBLDHXJMPZ "
+    inversion = "-1"
+    dawnglare_link = "https://pad.dawnglare.com/?patt={}&showfill=1"
+
+    def __init__(self, board):
+        self.board = board.split()
+        self.invalid_board = False
+        self.invert = self.inversion_check()
+        self.generate_clean_board()
+        self.validate_orbs()
+        self.board_length = len(self.clean_board)
+        self.eval_board_size()
+        self.generate_link()
+
+    def inversion_check(self):
+        if BoardGenerator.inversion in self.board:
+            invert = True
+        else:
+            invert = False
+
+        return invert
+
+    def generate_clean_board(self):
+        board = self.board
+
+        if self.invert:
+            board.remove(BoardGenerator.inversion)
+            clean = ''.join(self.invert_board(board))
+        else:
+            clean = ''.join(board)
+
+        self.clean_board = clean
+
+    def invert_board(self, board):
+        build_string = []
+        inverted_board = []
+
+        try:
+            for i in range(len(board[0])):
```

Comment:
I would put my local substring right here rather than at the top!  I would put `build_string = ""` and then `build_string += board[j][i]` in the loop right after!  That means there's no need for a second `clean_string` variable because our `build_string` is already a string. :)